lib/logstash/inputs/kinesis.rb

```diff
@@ -91,6 +94,14 @@ def register
       worker_id).
         withInitialPositionInStream(initial_position_in_stream).
         withRegionName(@region)
+
+      # Call arbitrary "withX()" functions
+      # replace snake_case => withCamelCase
+      # e.g. initial_position_in_stream => withInitialPositionInStream
+      @additional_kcl_options.each do |key, value|
+          fn = 'with' + key.title().replace('_', '')
```

Comment:
No need to do this - can replace this with `fn = "with_#{key}"`, snake case conversion to Java camelCase will work automatically

Reply:
Amazing. :mind-blown: 

Out of curiosity, what enables this magic? Something in ruby? jruby?

Reply:
Its a jruby thing - when calling a java method from ruby, either the original Java method name, or the equivalent idiomatic ruby method name can be used.